mod/thewire/actions/thewire/add.php

```diff
@@ -5,20 +5,48 @@
 
 // don't filter since we strip and filter escapes some characters
 $body = get_input('body', '', false);
+$guid = (int) get_input('guid');
 $parent_guid = (int) get_input('parent_guid');
 
 // make sure the post isn't blank
 if (empty($body)) {
 	return elgg_error_response(elgg_echo('thewire:blank'));
 }
 
-$guid = thewire_save_post($body, elgg_get_logged_in_user_guid(), ACCESS_PUBLIC, $parent_guid, 'site');
-if (!$guid) {
-	return elgg_error_response(elgg_echo('thewire:notsaved'));
-}
-
 $forward = null;
 
+$enable_editing = (bool) elgg_get_plugin_setting('enable_editing', 'thewire');
+
+if ($guid && $enable_editing) {
+	$entity = get_entity($guid);
+	
+	if (!$entity instanceof ElggWire) {
+		throw new \Elgg\EntityNotFoundException();
+	}
+	
+	if (!$entity->canEdit()) {
+		throw new \Elgg\EntityPermissionsException();
+	}
+	
+	$entity->description = $body;
+	
+	$save = $entity->save();
+	
+	if (!$save) {
```

Comment:
suggestion: since you don't re-use `$save` you could simply check `$entity->save()` in the `if` statement